compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -2591,8 +2590,8 @@ public void visit(BLangCollectClause collectClause) {
 
     void updateIdentifiers(SymbolEnv env) {
         for (Map.Entry<String, BSymbol> identifier : identifiers.entrySet()) {
-            BSymbol symbol = symResolver.lookupClosureVarSymbol(env, Names.fromString(identifier.getKey()),
-                    SymTag.SEQUENCE);
+            BSymbol symbol =
+                    symResolver.lookupSymbolInGivenScope(env, Names.fromString(identifier.getKey()), SymTag.SEQUENCE);
```

Comment:
There is a special case for the collect and group clauses of the query, where the type of the symbol is changed to a sequence (internal type used by queries). In this case, we have to obtain the sequence type symbol from the original symbol, and that's why we are using the name here.

Reply:
When defining a new symbol, we define that symbol within that scope. Could you provide an example where using a symbol fails to determine if the given symbol is within that scope?

Reply:
When reviewing the usages of the `lookupSymbolInGivenScope` function, it is good to use the symbol instead of the name of that symbol. Since this matter is not directly related to this PR, I have created a separate issue to address this improvement.https://github.com/ballerina-platform/ballerina-lang/issues/41886